### PR TITLE
Feature: Update Endpoint to guarantee cleanup (Endpoint.Start/Endpoint.Stop)

### PR DESCRIPTION
**Pull request by bfuzzy1**

### CODE REVIEW

File: endpoint/endpoint.go [hunk 1]
```diff
@@ -12,18 +12,21 @@ import (
...
-		println("[+] Starting cleanup")
-		RunWithTimeout(clean)
-	} else {
-		println("[+] Starting test")
-		RunWithTimeout(test)
-	}
+	cleanup = clean
+
+	println("[+] Starting cleanup")
+	RunWithTimeout(clean)
```
Comment (privateducky) [inline endpoint/endpoint.go:21]: running cleanup in the Start?

File: endpoint/endpoint.go [hunk 2]
```diff
@@ -12,18 +12,21 @@ import (
 
 type fn func()
 
+var cleanup fn = nil
+
 func Start(test fn, clean fn) {
```
Comment (privateducky) [inline endpoint/endpoint.go:17]: we should make ``clean`` an optional parameter, so a test can not pass it in and it'll use the default/anonymous cleanup defined in this module.

**Review by privateducky — CHANGES_REQUESTED** [review]
minor mods

File: endpoint/endpoint.go [hunk 3]
```diff
@@ -14,11 +14,15 @@ type fn func()
...
-func Start(test fn, clean fn) {
-	cleanup = clean
-
-	println("[+] Starting test")
-	RunWithTimeout(test)
+func Start(test fn, clean ...fn) {
+    if len(clean) > 0 {
+        cleanup = clean[0]
+    } else {
+        cleanup = func() {}
```
Comment (privateducky) [inline endpoint/endpoint.go]: don't need the ``else`` clause

Comment (bfuzzy1) [reply endpoint/endpoint.go]: wrong commit :D checkout tweak2 commit.

File: endpoint/endpoint.go [hunk 4]
```diff
@@ -12,18 +12,21 @@ import (
...
+
+func Start(test fn, clean ...fn) {
+    if len(clean) > 0 {
+        cleanup = clean[0]
+    }
+
+    println("[+] Starting test")
+    RunWithTimeout(test)
+
+    cleanup()
```
Comment (privateducky) [inline endpoint/endpoint.go:25]: why cleanup in the start and stop functions?

File: endpoint/endpoint.go [hunk 5]
```diff
@@ -12,18 +12,21 @@ import (
...
-	}
+var cleanup fn = func() {}
+
+func Start(test fn, clean ...fn) {
+    if len(clean) > 0 {
+        cleanup = clean[0]
+    }
+
+    println("[+] Starting test")
+    RunWithTimeout(test)
```
Comment (privateducky) [inline endpoint/endpoint.go:23]: given RunWithTimeout is now only used by the ``test``, we should pull that logic into the ``Start`` function

**Review by privateducky — APPROVED** [review]
please update the tests now, removing cleanup from those which don't actually use it.

**Comment by privateducky** [comment]
@bfuzzy1 note that after today's probe deployment, probes will now cleanup any files generated by running a test (including the test file itself) -- so tests that have clean up commands that only remove files can also be removed.

**Pull request merged**